beer-garden/functionary
# #90 - Initial combination of worker and listener into one entity

Closes #90 

This PR combines the `runner-worker` and `runner-listener` containers into a single container `functionary-runner`. The listener and worker are now `multiprocessing.Process` subclasses, each of which gets spun up from `runner.py`. Since Celery also uses `multiprocessing` for its workers, I configured Celery's logger to print all logging to stdout, so that they are visible when inspecting the container's logs. This avoids needing to do nested multiprocess logging.

## Introduced Changes
- Introduced new `runner.py` which spawns a `Worker` and `Listener` process
- The original functionality for `worker.py` has been turned into a `multiprocessing.Process` subclass
- The original functionality for `listener.py` has been turned into a `multiprocessing.Process` subclass
- Celery now streams its logs to `stdout` so the logs can be viewed when inspecting the container's logs

## Testing
- Spin up Functionary
- Inspect the `functionary-runner-listener` container's logs
  - Verify that listener is running (`INFO:runner.messaging:Connected to message broker`)
  - Verify the worker is running (`celery@... ready`)
- Execute a task from Functionary's UI
  - Verify task logs show up in the `functionary-runner-listener` container
  - Verify task runs as expected
